### ISSUE TITLE
[Call for Team] Future Development of sK1 & UniConvertor

### ISSUE THREAD

**Issue by Symbian9** [sk1project/sk1-wx#284]
**NOTE:** At the moment sK1 repo is active & maintained!

As for now, at least me has access to manage `sk1-wx` repo, BUT I has no access to other @sk1project's repos:

- https://github.com/sk1project?tab=repositories

**If some of you also has an access to mange this or other*** @sk1project's **repos, please, let me know!**

FYI @albertbronsky @maxim-s-barabash @k-bronski @sk1project-build-bot @raboof @coffee5 @AmbientLighter @THS-on @tomsv-gh @insolor

**Comment by maxim-s-barabash** [comment]
I'm here.

**Comment by Symbian9** [comment]
@maxim-s-barabash, please, check maybe you also has an access to manage `uniconvertor` repo (pin issues/approve pull requests/create release/etc.):

- https://github.com/sk1project/uniconvertor

**Comment by albertbronsky** [comment]
I will definitely check if I will be able to recover my father's GitHub account from his computer when I will have a chance to come back home from abroad. But that will be late May.

**Comment by AmbientLighter** [comment]
Alternatively, repository ownership may be transferred to successor via 
https://docs.github.com/en/github/site-policy/github-deceased-user-policy

Successor can manage repositories after presenting a death certificate then waiting for 7 days or presenting an obituary then waiting for 21 days.

**Comment by tomsv-gh** [comment]
Hi team, hi Albert.  I have just learned of what has happened and I am very saddened.  I have been following the project with great interest for years, initially because at the beginning of my career as a graphic designer I could not invest much money and Sk1 was a bit of a hope.  Later, even if I no longer needed it, I remained fond of the project and I have to thank Ihor who not only welcomed me into the team but often listened with interest to my advice and opinions, despite my not too much knowledge of software development.  Even if perhaps it will not be the same now, I will stay with the team and I will try in my small way to keep this project active and make it grow, as I am sure you will do too, so as to keep alive the memory of its founder.  RIP.

**Comment by albertbronsky** [comment]
FYI, 

- @k-bronski was my father's second account, which is now inactive.
- @sk1project-build-bot was his build bot.
- @coffee5 was my old account which I used to make contributions to English localization of sK1.

**Comment by insolor** [comment]
@Symbian9 
Hi
I made only one pull request to the [wixpy](https://github.com/sk1project/wixpy) project. I have no administrative right there.

**Comment by Symbian9** [comment]
> `@coffee5` was my old account which I used to make contributions to English

@albertbronsky, Has you an access to? (Just to check what the status in repos it has)

**Comment by coffee5** [comment]
> > `@coffee5` was my old account which I used to make contributions to English
> 
> @albertbronsky, Has you an access to old account? (Just to check what the status in repos it has)

Wow, I just checked this old account and it seems like it DOES have administrative rights for this repo (I can see that I can close issues, edit Readme, merge pull requests), but I am not sure to what extent?
My father never told me anything about granting me these privileges, I am quite surprised. 
I was helping him in 2017 with English localization of sK1 and he was specific back then about me not having any administrative rights here so I would not be able to mess up the repo, so I did a pull request back then and he merged it from his account.
Perhaps he could grant me these administrative rights while he was in the hospital while he still had his smartphone with him before being transferred to ICU? Is it possible to see an activity log here on GitHub?

Do you know how can I check what kind of permissions do I have in this repo?

Also, I checked the other repos on @sk1project account and unfortunately it seems I don't have any permissions for those, including UniConvertor.

**Comment by Symbian9** [comment]
> Perhaps he could grant me these administrative rights while he was in the hospital while he still had his smartphone with him before being transferred to ICU?

@coffee5 (@albertbronsky), I may just suppose that Ihor granted `Collaborator`'s right earlier, (as he did it for mostly of all of us during period of 2017-2019).

> Do you know how can I check what kind of permissions do I have in this repo?
>
>  Is it possible to see an activity log here on GitHub? 

Read GitHub Docs to see permissions list & difference between `Owner` and `Collaborator`:

- https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/permission-levels-for-a-user-account-repository

> Also, I checked the other repos on `@sk1project` account and unfortunately it seems I don't have any permissions for those, including UniConvertor.

At least @maxim-s-barabash has `Collaborator` access in both (`uniconvertor` and `sk1-wx`) repos.

**Comment by albertbronsky** [comment]
I don't have access to the official documents needed for transferring the project while I am abroad now. I will be coming home this summer and I will do it then.

Is `Collaborator` access enough for now to manage the repo?

**Comment by Symbian9** [comment]
> Is `Collaborator` access enough for now to manage the repo?

As for now its enough.

@albertbronsky, But transfer ownership to You is needed to take a full control (*better late than never*).

**Comment by albertbronsky** [comment]
> > Is `Collaborator` access enough for now to manage the repo?
> 
> As for now its enough.
> 
> @albertbronsky, But transfer ownership to You is needed to take a full control (_better late than never_).

I regained access to my father's account (@sk1project). What should I do now?

**Comment by tomsv-gh** [comment]
> > > Is `Collaborator` access enough for now to manage the repo?
> > 
> > As for now its enough.
> > 
> > @albertbronsky, But transfer ownership to You is needed to take a full control (_better late than never_).
> 
> I regained access to my father's account (@sk1project). What should I do now?

I thought about it for a while, there would be other things to recover (eg Facebook page, website) and maybe the solution could be to simply leave sk1 as it is and opt for a fork or at least a name change (Ihor?). 

I know that you have recently been busy changing the python version but I have also thought about a possible further complete rewrite in rust, for the future. 

In addition to this, I wish I could deal more with the visual side of the app and its visual communication. 

The last thing I would propose is to find a way to finance the project so that we can perhaps devote more time to it and motivate the team, perhaps with donations via Liberapay or something similar. 

In short, my idea is to try to make sK1 the definitive alternative to the most popular and commercial professional software of the same kind. I know they are huge changes but I thought about this for a long time before proposing them and in the end I decided to do it. I look forward to hearing your opinion on this.

**Comment by albertbronsky** [comment]
> > > > Is `Collaborator` access enough for now to manage the repo?
> > > 
> > > 
> > > As for now its enough.
> > > @albertbronsky, But transfer ownership to You is needed to take a full control (_better late than never_).
> > 
> > 
> > I regained access to my father's account (@sk1project). What should I do now?
> 
> I thought about it for a while, there would be other things to recover (eg Facebook page, website) and maybe the solution could be to simply leave sk1 as it is and opt for a fork or at least a name change (Ihor?).
> 
> I know that you have recently been busy changing the python version but I have also thought about a possible further complete rewrite in rust, for the future.
> 
> In addition to this, I wish I could deal more with the visual side of the app and its visual communication.
> 
> The last thing I would propose is to find a way to finance the project so that we can perhaps devote more time to it and motivate the team, perhaps with donations via Liberapay or something similar.
> 
> In short, my idea is to try to make sK1 the definitive alternative to the most popular and commercial professional software of the same kind. I know they are huge changes but I thought about this for a long time before proposing them and in the end I decided to do it. I look forward to hearing your opinion on this.

Yes, I have recovered Facebook and Twitter pages as well. I am now fixing the website now and I am waiting for domain transfer approval, so hopefully next week the site will be working again.

The name change would be a good idea, I guess. Maybe it can help revitalize the project.

The complete rewrite - I don't know if it's feasible with a small team. My father was working on migrating the project from Python 2 -> Python 3 lately. He was opposed to the idea of rewriting it to some low-level language, because he felt like the current model of writing logic in Python and occasionally using small C modules for interop in performance-critical parts is a better approach for a small team. I feel like rewriting the whole project in Rust is going to take many years for a team of few people.

Last year, my father started experimenting on a web-based version of sK1, with front-end written in JavaScript (something like Vectr, Pixlr, etc.). His vision was that vector editors would gradually move to browsers, since it's a more effective cross-platform solution that works without any installation at all. I found some initial work on his computer, as well as some notes on the future of the project. Maybe it's worth exploring this field?

**Comment by tomsv-gh** [comment]
> > > > > Is `Collaborator` access enough for now to manage the repo?
> > > > 
> > > > 
> > > > As for now its enough.
> > > > @albertbronsky, But transfer ownership to You is needed to take a full control (_better late than never_).
> > > 
> > > 
> > > I regained access to my father's account (@sk1project). What should I do now?
> > 
> > I thought about it for a while, there would be other things to recover (eg Facebook page, website) and maybe the solution could be to simply leave sk1 as it is and opt for a fork or at least a name change (Ihor?).
> > 
> > I know that you have recently been busy changing the python version but I have also thought about a possible further complete rewrite in rust, for the future.
> > 
> > In addition to this, I wish I could deal more with the visual side of the app and its visual communication.
> > 
> > The last thing I would propose is to find a way to finance the project so that we can perhaps devote more time to it and motivate the team, perhaps with donations via Liberapay or something similar.
> > 
> > In short, my idea is to try to make sK1 the definitive alternative to the most popular and commercial professional software of the same kind. I know they are huge changes but I thought about this for a long time before proposing them and in the end I decided to do it. I look forward to hearing your opinion on this.
> 
> Yes, I have recovered Facebook and Twitter pages as well. I am now fixing the website now and I am waiting for domain transfer approval, so hopefully next week the site will be working again.
> 
> The name change would be a good idea, I guess. Maybe it can help revitalize the project.
> 
> The complete rewrite - I don't know if it's feasible with a small team. My father was working on migrating the project from Python 2 -> Python 3 lately. He was opposed to the idea of rewriting it to some low-level language, because he felt like the current model of writing logic in Python and occasionally using small C modules for interop in performance-critical parts is a better approach for a small team. I feel like rewriting the whole project in Rust is going to take many years for a team of few people.
> 
> Last year, my father started experimenting on a web-based version of sK1, with front-end written in JavaScript (something like Vectr, Pixlr, etc.). His vision was that vector editors would gradually move to browsers, since it's a more effective cross-platform solution that works without any installation at all. I found some initial work on his computer, as well as some notes on the future of the project. Maybe it's worth exploring this field?



The browser variant is certainly interesting, even if being a software intended mainly for professionals, I believe that for most of them an installable software is more convenient, however the web version could find various uses on the customer side. Being able to implement the editor on a designer's website could dispose of a lot of minor work, when the client already knows what he wants and just needs to be guided in printing. If this would be possible, I think it will be amazing.

As for the fact that the team is small at the moment, I believe that by working on communication, making the project more known, others could join.  

As for another language, it is just a personal preference since we have sometimes found ourselves limited in the use of some graphic components (e.g. icon fonts for Windows) but since it's not to me to decide about the code, consider it just a proposal.

**Comment by Symbian9** [comment]
> The complete rewrite - I don't know if it's feasible with a small team. My father was working on migrating the project from Python 2 -> Python 3 lately. 

Yes, after *sK* `2.0` release development should be focused on migrating to Python 3.

> The browser variant is certainly interesting

Web version of *sK1* could be an additional version, but "classic" desktop variant of *sK1* should be primary version of this DTP (Desktop Publishing) software anyway.

**Comment by BharatKambalur** [comment]
I believe commit 08fcdd0cea179e7d7a9be2883cc6208454dafa75 addresses the first two TODOs. If a `CODEOWNERS` file is added, this issue could be closed and progress made on the milestone to v2.0. A newer [Call for Team] issue could be opened with more specifics on where help is needed and what blocking issues remain to release v2.0.
Additionally, if the conversion to python3 is planned for after release of v2.0, perhaps a new project could be opened to track the progress of tasks for it.
